File: baekjoon_all/02000+/boj_2004.cpp
// Solve 2022-06-08
// Update 2025-11-12

#include <bits/stdc++.h>

#define FASTIO ios_base::sync_with_stdio(false);cin.tie(NULL);
#define ALL(v) v.begin(),v.end()
#define UNIQUE(v) v.erase(unique(v.begin(),v.end()),v.end());
#define SETW(n, c) cout << setw(n) << setfill(c);
#define SETP(n) cout << fixed << setprecision(n);

using namespace std;
using ll = long long;
using uint = unsigned int;
using ull = unsigned long long;
using ld = long double;
using pii = pair<int, int>;
using pll = pair<ll, ll>;
const int INF = 0x3f3f3f3f;
const int MOD = 1000000007;

ll count_prime_in_factorial(ll n, ll p) {
    ll cnt = 0;

    while (n > 0) {
        n /= p;
        cnt += n;
    }

    return cnt;
}

int main() {
    FASTIO;

    ll n, m;
    cin >> n >> m;

    ll cnt2 = count_prime_in_factorial(n, 2);
    cnt2 -= count_prime_in_factorial(m, 2);
    cnt2 -= count_prime_in_factorial(n - m, 2);

    ll cnt5 = count_prime_in_factorial(n, 5);
    cnt5 -= count_prime_in_factorial(m, 5);
    cnt5 -= count_prime_in_factorial(n - m, 5);

    cout << min(cnt2, cnt5) << '\n';

    return 0;
}

File: baekjoon_all/02000+/boj_2004.py
# Solve 2023-10-09
# Update 2025-11-12

n, m = map(int, input().split())

cnt5, i = 0, 5

while i <= n:
    cnt5 += n // i
    cnt5 -= m // i
    cnt5 -= (n - m) // i

    i *= 5

cnt2, i = 0, 2

while i <= n:
    cnt2 += n // i
    cnt2 -= m // i
    cnt2 -= (n - m) // i

    i *= 2

print(min(cnt5, cnt2))
